### ISSUE TITLE
Address `DeprecationWarning` for joblib.Memory `cachedir` attribute

### ISSUE THREAD

**Issue by emdupre** [nilearn/nilearn#2329]
Currently, setting the `cachedir` attribute in, for example, `NiftiMasker` returns the following `DeprecationWarning`:

```python
/home/emdupre/miniconda3/envs/fmralign/lib/python3.7/site-packages/nilearn/_utils/cache_mixin.py:295: DeprecationWarning: The 'cachedir' attribute has been deprecated in version 0.12 and will be removed in version 0.14.
Use os.path.join(memory.location, 'joblib') attribute instead.
```

This is because we internally call [`joblib.Memory`](https://joblib.readthedocs.io/en/latest/auto_examples/memory_basic_usage.html#caching-the-result-of-a-function-to-avoid-recomputing), which has updated this argument from `cachedir` to `location` as of the 0.12 release. 

Given that the [latest joblib release is 0.14.1](https://pypi.org/project/joblib/0.14.1/), should we consider upgrading our internal calls ? If so, we should be able to fix this by also updating the attribute from `cachedir` to `location`; for example, here:

https://github.com/nilearn/nilearn/blob/ee1f25a908d8357b4fba84dab1d2b7018e9f5e04/nilearn/_utils/cache_mixin.py#L44

and here:

https://github.com/nilearn/nilearn/blob/ee1f25a908d8357b4fba84dab1d2b7018e9f5e04/nilearn/_utils/cache_mixin.py#L221

**Comment by bthirion** [comment]
Indeed !

**Comment by emdupre** [comment]
Great ! :tada: I agree with @GaelVaroquaux that this is a [Good first issue](https://github.com/nilearn/nilearn/labels/Good%20first%20issue).

To make it easier to approach, here are a few other things that will need to be updated:

The minimum tested version of joblib will need to be bumped up to 0.12 here:

https://github.com/nilearn/nilearn/blob/cc22775122dca01beda90626d63018dc4c82de38/.travis.yml#L27

and here:

https://github.com/nilearn/nilearn/blob/cc22775122dca01beda90626d63018dc4c82de38/.travis.yml#L34

and we'll need to update the minimum version for this dependency in `version.py`:

https://github.com/nilearn/nilearn/blob/9173aaca590df50bbaa6122bc744291617f36b90/nilearn/version.py#L45

Should we have a deprecation cycle for this, or just update at the next release ?

**Comment by GaelVaroquaux** [comment]
> Should we have a deprecation cycle for this, or just update at the next release ?

If the next release is 0.7, we should just upgrade, IMHO.

**Comment by kchawla-pi** [comment]
I believe I had opened a PR for this. It was stated until we had decided to
increase the minimum joblib version

**Comment by emdupre** [comment]
Thanks, @kchawla-pi ! I didn't check the closed PRs -- looks like you started to address this with https://github.com/nilearn/nilearn/pull/1946. From there (and the conversation above) it sounds like 0.7 is the release to drop 0.11 joblib support.

**Comment by emdupre** [comment]
Maybe @danjgale would be interested in looking into this, now that he's successfully finished his comps (:tada: :sparkles:)?

**Comment by kchawla-pi** [comment]
I will reopen the PR #1946 and another one related to numpy encoding and wrap them up, instead of opening a new PR.

Someone should look at #2315 .

@danjgale Congrats!!

**Comment by emdupre** [comment]
Closed with #1946
